### DISCUSSION TITLE
BallerinaMethodSymbol and  BallerinaSymbol

### DISCUSSION THREAD

**Discussion by dulajdilshan** [ballerina-platform/ballerina-lang#29778] [Q&A] [answered]
Is there any reason why we don't inherit `io.ballerina.compiler.api.impl.symbols.BallerinaMethodSymbol` from `io.ballerina.compiler.api.impl.symbols.BallerinaSymbol`?

Ref: https://github.com/ballerina-platform/ballerina-lang/blob/21d191b5ab5de7d3c4e25cc31b788ae8a034266b/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaMethodSymbol.java#L42

The reference issues for this are: #28388 #29467 

This happens in the `BallerinaSemanticModel` when we get the internal symbols, we have to go through this
https://github.com/ballerina-platform/ballerina-lang/blob/df065ef1142358b215cd8c9557545e0d2bdbd1b1/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java#L337

The code snippet:
```java
    private BSymbol getInternalSymbol(Symbol symbol) {
        if (symbol.kind() == TYPE) {
            return ((AbstractTypeSymbol) symbol).getBType().tsymbol;
        }

        return ((BallerinaSymbol) symbol).getInternalSymbol();
    }
```
Here, the error comes because `BallerinaMethodSymbol` is not inherited `BallerinaSymbol`' class.

@pubudu91 any thoughts?

**Comment by pubudu91** [comment, marked as answer]
Didn't extend from `BallerinaSymbol` at that time because there wasn't any particular reason to do so. The method symbol is basically a wrapper around a function symbol. We just simply call the function symbol's methods for each of the methods. This is for that NPE issue when looking up references of methods in classes right? No harm in extending from `BallerinaSymbol`. Since these are internal implementation details, we can change these as we see fit.

**Reply by dulajdilshan** [reply]
@pubudu91  Thanks. Yes, its referenced issues are updated in the discussion description. This is the change I have done: PR https://github.com/ballerina-platform/ballerina-lang/pull/29788. Please have a look and suggest any changes if I need to do
